### ISSUE TITLE
Batch Normalization Is Along Wrong Axis

### ISSUE THREAD

**Issue by HenryJia** [fchollet/keras#1921]
I've taken a look at Keras' batch normalization code and it looks like it's normalising along the wrong axis

https://github.com/fchollet/keras/blob/master/keras/layers/normalization.py#L64

This means that the axis parameter specifies the dimension to leave alone rather than the dimensions to compute the mean and std over

**Comment by fchollet** [comment]
It's very simple:

For `Dense` layer, all RNN layers and most other types of layers, the default of `axis=-1` is what you should use,
For `Convolution2D` layers with `dim_ordering="th"` (the default), use `axis=1`,
For `Convolution2D` layers with `dim_ordering="tf"`, use `axis=-1` (i.e. the default).

**Comment by fargiolas** [comment]
Sorry if I hijack this old issue but I don't want to open a new one for something that's probably be answered a tons of times. For some reason I missed the docstring about axis in BatchNormalization and I assumed that the image_data_format parameter  in keras.json would change all default axes where needed.

It's only when I started noticing vertical artifacts in my predictions, that I wondered what was wrong and found about the wrong axis parameter. Funnily enough my network was "learning around" the issue and only showed the artifacts when I stopped training too early.

Not sure if is there an easy solution for this, but I guess I'm not the only one who encountered it. I guess I'll just reshape my data to be in channel last ordering as I'm using tensorflow anyways.
